ping/rust/Dockerfile

```diff
@@ -4,43 +4,20 @@ WORKDIR /usr/src/testplan
 RUN apt-get update && apt-get install -y cmake protobuf-compiler
 
 ARG PLAN_PATH="./"
-
-# Cache dependencies between test runs, see testground examples.
-RUN mkdir -p ./plan/src/
-RUN echo "fn main() { println!(\"If you see this message, you may want to clean up the target directory or the Docker build cache.\") }" > ./plan/src/main.rs
-COPY ./plan/${PLAN_PATH}/Cargo.lock ./plan/${PLAN_PATH}/Cargo.toml ./plan/
-
-RUN cd ./plan/ && \
-        cargo update --dry-run && \
-        cargo fetch && \
-        # use a default feature to rely on docker caching.
-        cargo build --features "libp2pv0450"
-
-ARG CARGO_PATCH=""
-ARG CARGO_REMOVE=""
-
-RUN if [ ! -z "${CARGO_REMOVE}" ]; then sed -i "s/^${CARGO_REMOVE}.*//" ./plan/Cargo.toml; fi
-RUN echo "${CARGO_PATCH}" >> ./plan/Cargo.toml
-
-# Backup Cargo file to preserve patches.
-RUN cp ./plan/Cargo.toml ./plan/Cargo.lock /tmp/
-
 COPY ./plan/${PLAN_PATH} ./plan
+RUN cd ./plan/ && cargo build  # Initial build acts as a cache.
 
-# This is in order to make sure `main.rs`s mtime timestamp is updated to avoid the dummy `main`
-# remaining in the binary.
-# https://github.com/rust-lang/cargo/issues/9598
-RUN touch ./plan/src/main.rs
-
-RUN mv /tmp/Cargo.toml /tmp/Cargo.lock ./plan
+ARG GIT_TARGET=""
+RUN if [ ! -z "${GIT_TARGET}" ]; then export ESCAPED_GIT_TARGET="$(printf '%s' "${GIT_TARGET}" | sed -e 's/[\/]/\\&/g')"; sed -i "s/^git.*/git = \"https:\/\/${ESCAPED_GIT_TARGET}\"/" ./plan/Cargo.toml; fi # URLs contain forward slashes that need to be escaped for `sed` to work correctly.
```

Comment:
As an aside, you can use a different delimiter than `/` with `sed`, e.g. `#`.

See https://stackoverflow.com/a/5864155/4011134

Reply:
My god, knowing this would have saved me quite some time 😂

I'll try to simplify that then.